### PR TITLE
feat: refactor configs [JS-831]

### PR DESCRIPTION
**Pull request by shamsartem**

### CODE REVIEW

File: packages/cli/package/src/lib/lifeCycle.ts [hunk 1]
```diff
@@ -115,22 +87,16 @@ export async function initCli<
...
   if (requiresFluenceProject) {
-    setEnvConfig(await initNewEnvConfig());
+    await initNewEnvConfig();
   } else {
     try {
       // ensure env config also when provider.yaml exists
       await access(getProviderConfigPath());
-      setEnvConfig(await initNewEnvConfig());
-    } catch {
-      const envConfig = await initEnvConfig();
```
Comment (MixailE) [inline packages/cli/package/src/lib/lifeCycle.ts:128]: We don't need that anymore?

File: packages/cli/package/src/commands/local/up.ts [hunk 2]
```diff
@@ -97,48 +95,35 @@ export default class Up extends BaseCommand<typeof Up> {
...
   async run(): Promise<void> {
     const env: FluenceEnv = "local";
-
-    try {
-      const envConfig = await initNewEnvConfig(env);
-      envConfig.fluenceEnv = env;
-      await envConfig.$commit();
-      setEnvConfig(envConfig);
-    } catch (e) {
-      this.error(
```
Comment (MixailE) [inline packages/cli/package/src/commands/local/up.ts:107]: We don't need catch errors here anymore?

File: packages/cli/package/src/commands/provider/cc-collateral-withdraw.ts [hunk 3]
```diff
@@ -1,43 +0,0 @@
...
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-import { BaseCommand } from "../../baseCommand.js";
-import { collateralWithdraw } from "../../lib/chain/commitment.js";
-import { CHAIN_FLAGS, FLT_SYMBOL, CC_FLAGS } from "../../lib/const.js";
-import { initCli } from "../../lib/lifeCycle.js";
-
-// Deprecated alias for provider cc-finish
-export default class CCCollateralWithdraw extends BaseCommand<
```
Comment (MixailE) [inline packages/cli/package/src/commands/provider/cc-collateral-withdraw.ts:24]: We don't need `cc-collateral-withdraw`?

File: packages/cli/package/src/lib/lifeCycle.ts (hunk 1 above)
Comment (shamsartem) [reply packages/cli/package/src/lib/lifeCycle.ts:128]: I added a global "cache" for all configs long ago, cause essentially when you init a config object you can reuse it for the duration of CLI running cause it is expected to move anywhere, so there is no need anymore to store it globally separately. Finally removed that

File: packages/cli/package/src/commands/local/up.ts (hunk 2 above)
Comment (shamsartem) [reply packages/cli/package/src/commands/local/up.ts:107]: there previously was a check for env config that fluence.yaml exists. Now there is no such check cause e.g. providers also need env config, so there are no errors thrown there anymore, nothing to catch

File: packages/cli/package/src/commands/provider/cc-collateral-withdraw.ts (hunk 3 above)
Comment (shamsartem) [reply packages/cli/package/src/commands/provider/cc-collateral-withdraw.ts:24]: it still exists as an alias, it's an old naming, this file can be removed now after I updated oclif

**Review by MixailE — APPROVED** [review]
LGTM

**Pull request merged**